kv/memberlist/memberlist_client.go

```diff
@@ -932,6 +952,38 @@ func (m *KV) NodeMeta(limit int) []byte {
 func (m *KV) NotifyMsg(msg []byte) {
 	m.initWG.Wait()
 
+	// the passed message must be copied, since it may be altered after the call returns.
+	// https://github.com/hashicorp/memberlist/blob/master/delegate.go#L14-L15
+	msgBuff := m.notifyBuffPool.Get().(*bytes.Buffer)
+	io.Copy(msgBuff, bytes.NewReader(msg))
+
+	select {
+	case m.notifyCh <- msgBuff:
```

Comment:
This offloads processing from "notify" goroutine to background one, which is good, but doesn't exploit the structure of KV store to further parallelize processing of messages. We would prefer to have one channel and goroutine per key in KV store. If multiple goroutines are processing messages for the same key, they will need to serialize access to that key anyway.